### PR TITLE
#74845 Retain pipeline run

### PR DESCRIPTION
**Pull request by ru-ukho**

### CODE REVIEW

**Comment by srjames-ukho** [comment]
Hi Ravi, can you update the main README.md in the root directory to the direction to the README.md you added. Cheers

File: retain-pipeline/README.md
Comment (srjames-ukho) [inline retain-pipeline/README.md:1]: Hi Ravi, can you update the main README.md in the root directory to the direction to the README.md you added. Cheers

File: retain-pipeline/README.md
Comment (srjames-ukho) [inline retain-pipeline/README.md:24]:
```suggestion
- job:
```

File: retain-pipeline/README.md
Comment (srjames-ukho) [inline retain-pipeline/README.md:37]:
```suggestion
        dependsOn: "<DEPENDS ON JOB>"
        condition: "succeeded('<CONDITIONAL JOB>')"
        steps:  
           - checkout: UKHOTemplates
           - template: retain-pipeline/retain-pipeline.yml@UKHOTemplates
             parameters:
                DaysValid: 365
                AccessToken: "$(System.AccessToken)"
                DefinitionId: '$(System.DefinitionId)'
                OwnerId: '$(Build.RequestedForId)'
                BuildId: '$(Build.BuildId)'
                CollectionUri: '$(System.CollectionUri)'
                TeamProject: '$(System.TeamProject)'
```

File: retain-pipeline/README.md
Comment (srjames-ukho) [inline retain-pipeline/README.md:22]:
```suggestion
```

**Review by srjames-ukho — CHANGES_REQUESTED** [review]
Mostly some formatting in the example yaml. Could do with the root README.md updating with a reference to your new README.md file.

**Review by srjames-ukho — COMMENTED** [review]
Dismissing my request changes status as I will be annual leave next week and don't want that status to block the PR in my absense.

**Review by srjames-ukho — APPROVED** [review]

**Review by srjames-ukho — COMMENTED** [review]
Dismissed my request change status, should be safe for next week now

File: retain-pipelinerun/retain-pipelinerun.yml
Comment (elduddz) [inline retain-pipelinerun/retain-pipelinerun.yml:17]: please change how these are provided:
https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/reference/powershell-v2?view=azure-pipelines#call-powershell-script-with-multiple-arguments

File: retain-pipelinerun/retain-pipelinerun.yml
Comment (elduddz) [inline retain-pipelinerun/retain-pipelinerun.yml:11]: please add pwsh: true to help with use of powershell core.

**Review by elduddz — CHANGES_REQUESTED** [review]

**Review by elduddz — APPROVED** [review]

**Pull request merged**
